ldm/invoke/dynamic_prompts.py

```diff
@@ -0,0 +1,535 @@
+#!/usr/bin/env python
+
+"""
+Simple script to generate a file of InvokeAI prompts and settings
+that scan across steps and other parameters.
+"""
+
+import argparse
+import io
+import json
+import os
+import pydoc
+import re
+import shutil
+import sys
+import time
+from contextlib import redirect_stderr
+from io import TextIOBase
+from itertools import product
+from multiprocessing import Process
+from multiprocessing.connection import Connection, Pipe
+from pathlib import Path
+from tempfile import gettempdir
+from typing import Callable, Iterable, List
+
+import numpy as np
+import yaml
+from omegaconf import OmegaConf, dictconfig, listconfig
+
+
+def expand_prompts(
+    template_file: Path,
+    run_invoke: bool = False,
+    invoke_model: str = None,
+    invoke_outdir: Path = None,
+    processes_per_gpu: int = 1,
+):
+    """
+    :param template_file: A YAML file containing templated prompts and args
+    :param run_invoke: A boolean which if True will pass expanded prompts to invokeai CLI
+    :param invoke_model: Name of the model to load when run_invoke is true; otherwise uses default
+    :param invoke_outdir: Directory for outputs when run_invoke is true; otherwise uses default
+    """
+    if template_file.name.endswith(".json"):
+        with open(template_file, "r") as file:
+            with io.StringIO(yaml.dump(json.load(file))) as fh:
+                conf = OmegaConf.load(fh)
```

Comment:
🤸 those are some acrobatics. Have all those people who have been telling me "YAML is a superset of JSON" been lying to me?